### PR TITLE
issue #6: add basic endpoints app testing

### PR DESCRIPTION
**Pull request by rngadam**
issue #6 

success locally:

```
curl --data-urlencode arpTable@proc_net_arp http://127.0.0.1:3000/routerapi/v1/arp/candra -H "Authorization: Bearer abcd"
{"success":true,"message":"successfully retrieved data"}
```

### CODE REVIEW

**Comment by thimovss** [comment]
Rewrote,refactored,fixed and added testing to endpoint. 
Needs review.

File: app/imports/server/api/associateEntry.test.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    });
+  });
+
+  it('adds the entry if it is not present yet',function(){
+  	associateEntry(ExtraEntry);
+  	assert.equal(ARPEntries.find({'MAC': ExtraEntry.MAC}).count(), 1);
+  	assert.equal(ARPEntries.find().count(), 4);
+  });
+
+  it('doesn\'t add the the entry if it is present',function(){
```
Comment (rngadam) [inline app/imports/server/api/associateEntry.test.js]: doesn't -> does not, here and elsewhere

File: app/lib/collections/ARPEntries.js [hunk 2]
```diff
@@ -1,2 +1 @@
-App.Collections.ARPEntries = new Mongo.Collection('arptable');
-
+App.Collections.ARPEntries = new Mongo.Collection('arptable');
```
Comment (rngadam) [inline app/lib/collections/ARPEntries.js]: put back the newline

File: app/server/api/arpTable.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  rows.forEach((row, index) => {
+    if (index === 0)
+      return;
+    var values;
+    //remove double spaces
+    row = row.replace(/ +(?= )/g, '');
+    values = row.split(" ");
+
+    var IP;
+    var MAC;
```
Comment (rngadam) [inline app/server/api/arpTable.js]: uppercase variables should be lowercase

File: app/server/init.js [hunk 4]
```diff
@@ -20,14 +20,14 @@ function initOrgs() {
...
-      console.log(k);
+      //console.log(k);
       // Execute http calls asynchronously so we don't block execution for client inbound client request
       // Meteor.defer = Meteor.setTimeout(x, 0)
       var url = directory.data[k];
 
       Meteor.defer(function () {
         try {
-          console.log('fetching ' + url);
+          //console.log('fetching ' + url);
```
Comment (rngadam) [inline app/server/init.js]: revert these changes as they'll conflict with the other pull requests changes

File: app/server/regexes.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+var isIP = function(field) {
+    return new RegExp("^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$").test(field);
+};
+
+var isMAC = function(field) {
+    return new RegExp("^([0-9A-F]{2}[:-]){5}([0-9A-F]{2})$").test(field);
+};
+
+export {isIP, isMAC};
```
Comment (rngadam) [inline app/server/regexes.js]: newline

File: app/server/regexes.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+var isIP = function(field) {
+    return new RegExp("^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$").test(field);
```
Comment (rngadam) [inline app/server/regexes.js]: expensive!

Comment (rngadam) [reply app/server/regexes.js]: remove this file

File: app/server/api/endpoints.js [hunk 7]
```diff
@@ -1,129 +1,97 @@
...
+            if (!headers.authorization) {
+                return {
+                    statusCode: 401,
+                    body: {
+                        success: false,
+                        message: 'authorization token missing'
+                    }
+                };
+            }
+
```
Comment (rngadam) [inline app/server/api/endpoints.js:39]: extra whiteline

File: app/server/api/arpTable.js [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    // A new user connected to the router
+    // If MAC address doesn't exist in mongodb, add it
+    if (!ARPEntry) {
+
+      var response;
+
+      try {
+        response = HTTP.call("POST", "http://api.macvendors.com/" + entry.MAC);
+      } catch (e) {
+        // Got a network error, time-out or HTTP error in the 400 or 500 range.
```
Comment (rngadam) [inline app/server/api/arpTable.js:69]: create issue to log failed http requests to third party API like this
